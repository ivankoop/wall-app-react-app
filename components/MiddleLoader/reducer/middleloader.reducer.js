import {
    SHOW_MIDDLE_LOADER,
    HIDE_MIDDLE_LOADER,
} from "../constants/middleloader.constants";

const INTIAL_STATE = {
    visibleLoader: false,
};

export function middleLoaderControls(state = INTIAL_STATE, action) {
    switch (action.type) {
        case SHOW_MIDDLE_LOADER:
            return {
                ...state,
                visibleLoader: true,
            };
        case HIDE_MIDDLE_LOADER:
            return {
                ...state,
                visibleLoader: false,
            };

        default:
            return state;
    }
}
