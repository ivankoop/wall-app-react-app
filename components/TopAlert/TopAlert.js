import React, {Component} from "react";
import {View, Text, StyleSheet, Animated, StatusBar, Image} from "react-native";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {isIphoneX} from "../../helpers/Utils";
import {hideTopAlert} from "./actions/topalert.actions";

class TopAlert extends Component {
    constructor(props) {
        super(props);

        this.state = {
            animation: new Animated.Value(0),
            animationRunning: false,
        };
    }

    static defaultProps = {
        type: "warning",
        text: "Ha ocurrido un error.",
    };

    static propTypes = {
        type: PropTypes.string,
        text: PropTypes.string,
        visible: PropTypes.bool,
        hideTopAlert: PropTypes.func,
        alertType: PropTypes.string,
    };

    _showAlert = () => {
        const {hideTopAlert} = this.props;

        Animated.timing(this.state.animation, {
            toValue: isIphoneX() ? 130 : 80,
            duration: 200,
        }).start();

        setTimeout(async () => {
            await Animated.timing(this.state.animation, {
                toValue: 0,
                duration: 200,
            }).start(() => {
                hideTopAlert();
            });
        }, 2000);
    };

    componentDidUpdate() {
        const {visible} = this.props;
        if (visible) {
            this._showAlert();
        }
    }

    render() {
        const {animation} = this.state;
        const {alertType, text, visible} = this.props;

        let icon = iconWarningSmall;

        if (alertType === "error") {
            icon = iconErrorSmall;
        } else if (alertType === "success") {
            icon = iconCheckSmall;
        }

        return (
            visible && (
                <Animated.View
                    style={{...styles.mainContainer, height: animation}}
                >
                    <StatusBar
                        barStyle="light-content"
                        backgroundColor="transparent"
                    />
                    <View style={styles.contentContainer}>
                        <Image source={icon} style={styles.icon}></Image>
                        <Text style={styles.text}>{text}</Text>
                    </View>
                </Animated.View>
            )
        );
    }
}

const mapStateToProps = state => {
    const {visible, alertType, text} = state.topAlertControls;

    return {
        visible: visible,
        alertType: alertType,
        text: text,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        hideTopAlert: () => dispatch(hideTopAlert()),
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(TopAlert);

const iconCheckSmall = require("../../assets/images/icon-check-small.png");
const iconErrorSmall = require("../../assets/images/icon-error-small.png");
const iconWarningSmall = require("../../assets/images/icon-warning-small.png");

const styles = StyleSheet.create({
    mainContainer: {
        position: "absolute",
        top: 0,
        zIndex: 25,
        height: 0,
        width: "100%",
        backgroundColor: "rgba(0,0,0,0.8)",
    },

    contentContainer: {
        flexDirection: "row",
        paddingLeft: 19,
        paddingRight: 19,
        alignItems: "center",
        position: "absolute",
        bottom: 25,
    },

    icon: {
        width: 24,
        height: 24,
        marginRight: 11,
    },

    text: {
        color: "white",
        fontSize: 12,
        marginRight: 20,
        fontFamily: "Roboto_medium",
    },
});
