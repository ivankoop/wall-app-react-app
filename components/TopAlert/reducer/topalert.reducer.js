import {SHOW_TOP_ALERT, HIDE_TOP_ALERT} from "../constants/topalert.constants";

const INTIAL_STATE = {
    alertType: "warning",
    text: "",
    visible: false,
};

export function topAlertControls(state = INTIAL_STATE, action) {
    switch (action.type) {
        case SHOW_TOP_ALERT:
            return {
                ...state,
                alertType: action.alertType,
                text: action.text,
                visible: true,
            };
        case HIDE_TOP_ALERT:
            return {
                ...state,
                alertType: "warning",
                text: "",
                visible: false,
            };

        default:
            return state;
    }
}
