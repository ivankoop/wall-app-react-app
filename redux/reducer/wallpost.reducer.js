import {
    GET_WALL_POST_REQUEST,
    GET_WALL_POST_SUCCESS,
    GET_WALL_POST_FAILURE,
    INSERT_WALL_POST,
    DELETE_WALL_POST,
} from "../constants/wallpost.constants";

const INTIAL_STATE = {
    wallPosts: [],
    wallPostsLoading: false,
};

export function wallPostsControls(state = INTIAL_STATE, action) {
    switch (action.type) {
        case GET_WALL_POST_REQUEST:
            return {
                ...state,
                wallPostsLoading: true,
            };
        case GET_WALL_POST_SUCCESS:
            return {
                ...state,
                wallPostsLoading: false,
                wallPosts: action.wallPosts,
            };

        case GET_WALL_POST_FAILURE:
            return {
                ...state,
                wallPostsLoading: false,
            };

        case INSERT_WALL_POST:
            return {
                ...state,
                wallPosts: [action.post].concat(state.wallPosts),
            };

        case DELETE_WALL_POST:
            return {
                ...state,
                wallPosts: state.wallPosts.filter(
                    ({_id}) => _id !== action.postId
                ),
            };

        default:
            return state;
    }
}
