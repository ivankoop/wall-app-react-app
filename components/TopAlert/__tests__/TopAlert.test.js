import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {combineReducers, createStore} from "redux";
import TopAlert from "../TopAlert";
import {topAlertControls} from "../reducer/topalert.reducer";

const myReducers = combineReducers({
    topAlertControls,
});

const store = createStore(myReducers);

describe("TopAlert", () => {
    beforeEach(() => {
        console.error = jest.fn();
    });

    it("renders TopAlert", () => {
        jest.useFakeTimers();
        const tree = renderer
            .create(
                <Provider store={store}>
                    <TopAlert visible={true} />
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
