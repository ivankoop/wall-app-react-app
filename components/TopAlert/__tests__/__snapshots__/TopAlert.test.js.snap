// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAlert renders TopAlert 1`] = `null`;
